networkx/algorithms/shortest_paths/ada_star.py

```diff
@@ -0,0 +1,470 @@
+    return a path that is suboptimal by this bound. The algorithm
+    will return a path that is suboptimal by this bound. The path
+    is accessable through the extract_path function.
+
+
+    Methods
+    -------
+    compute_or_improve_path(epsilon)
+        Compute or improve the path from the start state to the goal state.
+
+    update_graph(changes)
+        Update the graph with new edge weights
+
+    extract_path()
+        Extract the path based on current potentially suboptimal solution.
+
+
+    Raises
+    ------
+    NetworkXNoPath
+        If no path exists between source and target.
+    NodeNotFound
+        If either source or target is not in G.
+
+    See Also
+    --------
+    shortest_path, astar_path, dijkstra_path
+
+    References
+    ----------
+    [1] Likhachev, Maxim, David I. Ferguson, Geoffrey J. Gordon, Anthony
+    Stentz, and Sebastian Thrun. "Anytime dynamic A*: An anytime,
+    replanning algorithm." In ICAPS, vol. 5, pp. 262-271. 2005.
+
+    Examples
+    --------
+    >>> import math
+
+    >>> G = nx.random_geometric_graph(100, 0.20, seed=896803)
+    >>> for u, v, w in G.edges(data=True):  # Euclidean distance between nodes
+    ...     w["weight"] = math.sqrt(
+    ...         (G.nodes[v]["pos"][0] - G.nodes[u]["pos"][0]) ** 2
+    ...         + (G.nodes[v]["pos"][1] - G.nodes[u]["pos"][1]) ** 2
+    ...     )
+    >>> source, target = 42, 25
+
+    >>> def heursistic(u, v):  # Euclidean distance between nodes
+    ...     return math.sqrt(
+    ...         (G.nodes[v]["pos"][0] - G.nodes[u]["pos"][0]) ** 2
+    ...         + (G.nodes[v]["pos"][1] - G.nodes[u]["pos"][1]) ** 2
+    ...     )
+
+    >>> # A* search for comparison
+    >>> path = nx.astar_path(G, source, target, heursistic)
+    >>> print("A* path: ", path)
+    A* path:  [42, 32, 19, 72, 49, 29, 31, 94, 35, 25]
+    >>> search = ada_star(source, target, G, heursistic)
+
+    >>> search.compute_or_improve_path(epsilon=2)
+    >>> path = search.extract_path()
+    >>> print("epsilon = 2 path: ", path)
+    epsilon = 2 path:  [42, 32, 24, 40, 59, 4, 66, 27, 35, 25]
+    >>> print(
+    ...     "epsilon = 2 path_weight: ", nx.path_weight(G, search.extract_path(), "weight")
+    ... )
+    epsilon = 2 path_weight:  1.4679609830956495
+
+    >>> search.compute_or_improve_path(epsilon=1.2)
+    >>> path = search.extract_path()
+    >>> print("epsilon = 1.2 path: ", path)
+    epsilon = 1.2 path:  [42, 32, 24, 12, 59, 4, 1, 27, 35, 25]
+    >>> print("epsilon = 1.2 path_weight: ", nx.path_weight(G, path, "weight"))
+    epsilon = 1.2 path_weight:  1.3335657361796027
+
+    >>> search.compute_or_improve_path(epsilon=1)
+    >>> path = search.extract_path()
+    >>> print("epsilon = 1 path: ", path)
+    epsilon = 1 path:  [42, 32, 19, 72, 49, 29, 31, 94, 35, 25]
+    >>> print("epsilon = 1 path_weight: ", nx.path_weight(G, path, "weight"))
+    epsilon = 1 path_weight:  1.29129785933092
```

Comment:
These examples would be more readable with less text.  Would something like this work:
```python
    >>> path = search.extract_path()
    >>> path, nx.path_weight(G, path, "weight")
    [42, 32, 19, 72, 49, 29, 31, 94, 35, 25], 1.29129785933092

Reply:
I have made your suggested changes to the examples! thanks again